### DISCUSSION TITLE
Shall we store Valid.Dead move result to move stack?

### DISCUSSION THREAD

**Discussion by EDChui** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#21] [Q&A] [answered]
Hello.
In the javadoc of `MoveStack.java`, it mentioned that `MoveStack` is used to track **_all valid_** moves made by a player.
However, according to the FAQ, it mention that `Valid.Dead` are not undoable,.

So shall we still store the `Valid.Dead` move to the stack? I also notice that when I store the it to the stack, so test cases failed. 
Thank you!

**Comment by Derppening** [comment, marked as answer]
The point of the `MoveStack` is for undoing, so if `Valid.Dead` is not undoable, you should not push `Valid.Dead` to the stack.

Alternatively, you may push the move onto the stack in your implementation, but when `GameController.processUndo` returns, the move should have been popped. Some test cases will catch this for you.

**Reply by EDChui** [reply]
Thanks for the clarification!
